src/fsharp/CompilerConfig.fs

```diff
@@ -934,6 +934,86 @@ type TcConfig private (data: TcConfigBuilder, validate: bool) =
+                        yield runtimeRootWPF // PresentationCore.dll is in C:\Windows\Microsoft.NET\Framework\v4.0.30319\WPF
+
+                    match data.FxResolver.GetFrameworkRefsPackDirectory() with
+                    | Some path when FileSystem.DirectoryExistsShim(path) ->
+                        yield path
+                    | _ -> ()
+
+                | LegacyResolutionEnvironment.EditingOrCompilation _ ->
+#if ENABLE_MONO_SUPPORT
+                    if runningOnMono then
+                        // Default compilation-time references on Mono
+                        //
+                        // On Mono, the default references come from the implementation assemblies.
+                        // This is because we have had trouble reliably using MSBuild APIs to compute DotNetFrameworkReferenceAssembliesRootDirectory on Mono.
+                        yield runtimeRoot
+                        if FileSystem.DirectoryExistsShim runtimeRootFacades then
+                            yield runtimeRootFacades // System.Runtime.dll is in /usr/lib/mono/4.5/Facades
+                        if FileSystem.DirectoryExistsShim runtimeRootWPF then
+                            yield runtimeRootWPF // PresentationCore.dll is in C:\Windows\Microsoft.NET\Framework\v4.0.30319\WPF
+                        // On Mono we also add a default reference to the 4.5-api and 4.5-api/Facades directories.
+                        let runtimeRootApi = runtimeRootWithoutSlash + "-api"
+                        let runtimeRootApiFacades = Path.Combine(runtimeRootApi, "Facades")
+                        if FileSystem.DirectoryExistsShim runtimeRootApi then
+                            yield runtimeRootApi
+                        if FileSystem.DirectoryExistsShim runtimeRootApiFacades then
+                             yield runtimeRootApiFacades
+                    else
+#endif
+                        // Default compilation-time references on .NET Framework
+                        //
+                        // This is the normal case for "fsc.exe a.fs". We refer to the reference assemblies folder.
+                        let frameworkRoot = data.legacyReferenceResolver.Impl.DotNetFrameworkReferenceAssembliesRootDirectory
+                        let frameworkRootVersion = Path.Combine(frameworkRoot, targetFrameworkVersionValue)
+                        yield frameworkRootVersion
+                        let facades = Path.Combine(frameworkRootVersion, "Facades")
+                        if FileSystem.DirectoryExistsShim facades then
+                            yield facades
+                        match data.FxResolver.GetFrameworkRefsPackDirectory() with
+                        | Some path when FileSystem.DirectoryExistsShim(path) ->
+                            yield path
+                        | _ -> ()
+                  ]
+        with e ->
+            errorRecovery e range0; []
```

Comment:
how does this behave when the `unwindBuildPhase` step from the previous version of this code isn't present?

Reply:
This was moved to the `TcConfig` constructor and the phase is present when it's being created:
https://github.com/dotnet/fsharp/blob/be1cbb7f9f4e28340bf5a77cba33eec069db3d7b/src/fsharp/CompilerConfig.fs#L1067-L1069

Reply:
Thanks, that makes sense.